R/ji_completion.R

```diff
@@ -22,3 +22,34 @@ print.emoji_completion <- function(x, ...){
   writeLines( paste(x, " : ", names(x)) )
   invisible(x)
 }
+
+emoji_completer <- function(env) {
+  if (!completeme::inside_quotes(env) || !grepl(":[^[:space:]]*:?$", env$token)) {
+    return()
+  }
+
+  # The IDE does not tokenize the words like the console
+  token <- sub("[^:]*:", ":", env$token)
+
+  if (grepl("^:[^[:space:]]+:$", token)) {
+    emoji <- paste0(sub(":[^[:space:]]+:$", "", env$token), emo::ji(gsub(":", "", token)))
+  } else {
+    res <- ji_completion(sub(":", "", token))
+
+    if (length(res) == 0) {
+      return()
+    }
+
+    if (length(res) == 1) {
+      emoji <- emo::ji(names(res))
+    } else {
+      emoji <- paste0(res, " : ", names(res))
+    }
+  }
```

Comment:
The popup would look better if `emoji_completer` was allowed to return a list (similar to what `.rs.makeCompletions` makes, and we would not need to special case `if (length(res) == 1)`. 

or perhaps a string (for compatibility with the console) and some extra information, e.g. in attributes, when used within rstudio. 

This would need a minor change to the IDE, I can send a small PR.